lib/house.rb

```diff
@@ -0,0 +1,45 @@
+
+class House
+  attr_reader :price,
+              :address,
+              :rooms,
+              :area
+
+  def initialize(price, address)
+    @price = price
+    @address = address
+    @rooms = []
+  end
+
+  def add_room(room)
+    @rooms << room
+  end
+
+  def above_market_average?
+    if @price.to_i > 500000
+      true
+    else
+      false
+    end
+  end
+
+  def rooms_from_category(category)
+    @rooms
+  end
```

Comment:
Anytime we need to work with a collection of things (`@rooms`) in this case, the first tool I turn to is iteration, either with `each` or another enumerable.  In this case, either of these solutions would be great:

```ruby
  def rooms_from_category(category)
    found_rooms = []
    @rooms.find_all do |room|
      if room.category == category
       found_rooms << room
      end
    end
    found_rooms
  end
```
OR
```ruby
  def rooms_from_category(category)
    @rooms.find_all do |room|
      room.category == category
    end
  end
```